src/yeoda/datacube.py

```diff
@@ -1,304 +1,146 @@
-from yeoda.utils import any_geom2ogr_geom
 from yeoda.utils import to_list
-from yeoda.utils import swap_axis
-from yeoda.utils import get_polygon_envelope
+from yeoda.utils import create_fn_class
+
 
 # load classes from yeoda's error module
-from yeoda.errors import IOClassNotFound
-from yeoda.errors import DataTypeUnknown
-from yeoda.errors import TileNotAvailable
-from yeoda.errors import FileTypeUnknown
 from yeoda.errors import DimensionUnkown
-from yeoda.errors import LoadingDataError
-from yeoda.errors import SpatialInconsistencyError
-
-
-def _check_inventory(f):
-    """
-    Decorator for `EODataCube` functions to check if the inventory exists.
 
-    Parameters
-    ----------
-    f : function
-        'EODataCube' function that has a keyword argument `inplace`
 
-    Returns
-    -------
-    function
-        Wrapper around `f`.
-    """
-
-    def f_wrapper(self, *args, **kwargs):
-        inplace = kwargs.get('inplace')
-        if self.inventory is not None:
-            return f(self, *args, **kwargs)
-        else:
-            if inplace:
-                return None
-            else:
-                return self
-    return f_wrapper
-
-
-def _set_status(status):
-    """
-    Decorator for `EODataCube` functions to set internal flag defining if a process changes the data cube structure or
-    not.
-
-    Parameters
-    ----------
-    status: str
-        Flag defining the status of the data cube. It can be:
-            - "changed": a filtering process was executed, therefore the structure of the data cube has changed.
-            - "stable": the structure of the data cube is remains the same.
-
-    Returns
-    -------
-    function
-        Wrapper around `f`.
-    """
-
-    def decorator(f):
-        def f_wrapper(self, *args, **kwargs):
-            ret_val = f(self, *args, **kwargs)
-            inplace = kwargs.get('inplace', None)
-            if inplace:
-                self.status = status
-            return ret_val
-        return f_wrapper
-    return decorator
-
-
-class EODataCube:
-    """
-    A file(name) based data cube for preferably gridded and well-structured EO data.
-
-    """
-
-    def __init__(self, filepaths=None, grid=None, filename_class=None, dimensions=None, inventory=None,
-                 io_map=None, sdim_name="tile", tdim_name="time"):
-        """
-        Constructor of the `EODataCube` class.
+FILE_CLASS = {'.tif': GeoTiffFile,
+              '.nc': NetCdf4File}
+RASTER_DATA_CLASS = {'.tif': (GeoTiffReader, GeoTiffWriter),
+                     '.nc': (NetCdfReader, NetCdfWriter)}
+PROC_OBJS = {}
+
+
+def parse_init(filepaths, fn_class, file_class, fn_dims, md_dims, md_decoder, tmp_dirpath):
+    PROC_OBJS['filepaths'] = filepaths
+    PROC_OBJS['filename_class'] = fn_class
+    PROC_OBJS['file_class'] = file_class
+    PROC_OBJS['fn_dims'] = fn_dims
+    PROC_OBJS['md_dims'] = md_dims
+    PROC_OBJS['md_decoder'] = md_decoder
+    PROC_OBJS['tmp_dirpath'] = tmp_dirpath
```

Comment:
can we maybe get away without mutable globals, this might create issues when processes run in parallel

Reply:
Lets create an issue for this and fix this later.